cryptoki/src/object.rs

```diff
@@ -617,7 +617,7 @@ impl Attribute {
             | Attribute::Verify(_)
             | Attribute::VerifyRecover(_)
             | Attribute::Wrap(_)
-            | Attribute::WrapWithTrusted(_) => 1,
+            | Attribute::WrapWithTrusted(_) => std::mem::size_of::<bool>(),
```

Comment:
Extreme corner case -> `CK_BBOOL` is defined as an `unsigned char`.  What if the architecture defines a `char` to not be 8-bits (e.g. not POSIX compliant).

Googling this question turns up two TI DSPs that if someone implemented a cryptoki token on I'd question their sanity.

I think this is fine, but just wanted to pose the question.

Reply:
Good catch. It may be possible at an architectural level, but the standard itself excludes that possibility by explicitly saying "unsigned 8-bit value".

At this particular location, the type being sized is exactly the type that the raw pointer points to. So if, say, CK_BBOOL somehow gets bigger because of a weird backing c_uchar, the ptr and len calls still match each other. And the provider should return `CKR_BUFFER_TOO_SMALL` or some other error code.